policy-controller/src/lib.rs

```diff
@@ -136,14 +147,35 @@ impl DiscoverOutboundPolicy<OutboundDiscoverTarget> for OutboundDiscover {
         port: NonZeroU16,
         source_namespace: String,
     ) -> Option<OutboundDiscoverTarget> {
+        // first try to lookup a service that we have indexed
+        if let Some(outbound::ResourceRef { name, namespace }) = self.0.read().lookup_service(addr)
```

Comment:
It would be probably be better than to take the read lock once and use it throughout the function instead of taking the read lock up to 3 times.